java-cfenv/src/main/java/io/pivotal/cfenv/core/CfVolume.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pivotal.cfenv.core;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+/**
+ * Access individual volume.
+ *
+ * @author Paul Warren
+ */
+public class CfVolume {
+
+	private static final String READ_ONLY = "r";
+
+	private final Map<String, String> volumeData;
```

Comment:
initialize to empty  map to avoid a possible NPE.

Reply:
Since it is final and that matched the constructor arg pattern in `CfCredential` we decide to check this in the constructor instead.  Also concluded that Spring in general (and this project)  already throw RuntimeException's and therefore used `Objects.requireNotNull` to do that check.